cin_validator/rules/cin2022_23/rule_8670Q.py

```diff
@@ -0,0 +1,168 @@
+"""
+Rule number: 8670Q
+Module: Assessments
+Rule details: Where present, within an <Assessments> group, if <AssessmentAuthorisationDate> (N00160) is not present then <AssessmentActualStartDate>
+(N00159) should not be before the <ReferenceDate> (N00603) minus 45 working days.
+
+Rule message: Please check: Assessment started more than 45 working days before the end of the census year. However, there is no Assessment end date. 
+
+"""
+from typing import Mapping
+
+import pandas as pd
+import datetime as dt
+
+from cin_validator.rule_engine import (
+    CINTable,
+    IssueLocator,
+    RuleContext,
+    RuleType,
+    rule_definition,
+)
+from cin_validator.test_engine import run_rule
+from cin_validator.utils import make_census_period
+
+# Get tables and columns of interest from the CINTable object defined in rule_engine/__api.py
+
+Assessments = CINTable.Assessments
+AssessmentAuthorisationDate = Assessments.AssessmentAuthorisationDate
+AssessmentActualStartDate = Assessments.AssessmentActualStartDate
+
+Header = CINTable.Header
+ReferenceDate = Header.ReferenceDate
+
+# define characteristics of rule
+@rule_definition(
+    code="8670Q",
+    module=CINTable.Assessments,
+    rule_type=RuleType.QUERY,
+    message="Please check: Assessment started more than 45 working days before the end of the census year. However, there is no Assessment end date. ",
+    affected_fields=[AssessmentActualStartDate],
+)
+def validate(
+    data_container: Mapping[CINTable, pd.DataFrame], rule_context: RuleContext
+):
+    df_assessments = data_container[Assessments]
+    df_assessments.index.name = "ROW_ID"
+
+    df_ref = data_container[Header]
+    ref_date_series = df_ref[ReferenceDate]
+    collection_start, collection_end = make_census_period(ref_date_series)
+
+    #  If <AssessmentAuthorisationDate> (N00160) is not present then <AssessmentActualStartDate> (N00159) should not be before the <ReferenceDate> (N00603) minus 45 working days
+
+    # Filter to only those with no authorisation date
+    df_assessments = df_assessments[df_assessments[AssessmentAuthorisationDate].isna()]
+
+    # Find the reference date - 45
+    latest_date = collection_end - pd.tseries.offsets.BDay(45)
+
+    df_issues = df_assessments[
+        df_assessments[AssessmentActualStartDate] < latest_date
+    ].reset_index()
+
+    link_id = tuple(
+        zip(
+            df_issues[AssessmentAuthorisationDate], df_issues[AssessmentActualStartDate]
```

Comment:
Okay, so if the code checks that assessment authorisation data .isna(), then this makes the first part of this link ID a NaN, so LAchildID might be better!